android-opencv/build.sh

```diff
@@ -0,0 +1,86 @@
+#!/usr/bin/env bash
+# ==================================================================================================
+# Copyright (c) 2018, Jairus Martin.
+# Copyright 2018 by Rodrigo Gomes <rodgomesc@gmail.com>
+# Distributed under the terms of the GPL v3 License.
+# The full license is in the file LICENSE, distributed with this software.
+# Created on Nov 22, 2018
+# ==================================================================================================
+
+# need Android SDK Tools to version 25.2.5
+# Details: https://github.com/opencv/opencv/issues/8460
+
+export ARCHS=("x86_64 x86 arm arm64")
+export NDK="$HOME/Android/Sdk/ndk-bundle"
+export ANDROID_SDK_ROOT="$HOME/Android/Sdk"
+
+for ARCH in $ARCHS
+do
+
+    CONFIGURE_FLAGS=""
+    if [ "$ARCH" == "arm" ]; then
+        export TARGET_HOST="arm-linux-androideabi"
+        export TARGET_ABI="armeabi-v7a"
+    elif [ "$ARCH" == "arm64" ]; then
+        export TARGET_HOST="aarch64-linux-android"
+        export TARGET_ABI="arm64-v8a"
+        
+    elif [ "$ARCH" == "x86" ]; then
+        export TARGET_HOST="i686-linux-android"
+        export TARGET_ABI="x86"
+    elif [ "$ARCH" == "x86_64" ]; then
+        export TARGET_HOST="x86_64-linux-android"
+        export TARGET_ABI="x86_64"
+    fi
+
+    export ANDROID_TOOLCHAIN="$NDK/standalone/$ARCH"
+    export APP_ROOT="$PREFIX/android/$ARCH"
+    export PATH="$PATH:$ANDROID_TOOLCHAIN/bin"
+    export AR="$TARGET_HOST-ar"
+    #export AS="$TARGET_HOST-clang"
+    export CC="$TARGET_HOST-clang"
+    export CXX="$TARGET_HOST-clang++"
+    export LD="$TARGET_HOST-ld"
+    
+    if [ "${TARGET_ABI}" = "armeabi" ]; then
+        API_LEVEL=19
+    else
+        API_LEVEL=21
+    fi
+
+    rm -Rf build || true
+    mkdir build
+    cd build       
+	cmake .. -DCMAKE_BUILD_WITH_INSTALL_RPATH=ON \
+	      -DCMAKE_TOOLCHAIN_FILE="$NDK/build/cmake/android.toolchain.cmake" \
+              -DANDROID_ARM_NEON=ON \
+	      -DANDROID_NDK="${NDK}" \
+	      -DANDROID_NATIVE_API_LEVEL=${API_LEVEL} \
+	      -DANDROID_ABI="${ANDROID_ABI}" \
+              -DCMAKE_FIND_ROOT_PATH="$ANDROID_TOOLCHAIN/sysroot" \
+              -DCMAKE_C_COMPILER="$TARGET_HOST-clang" \
+              -DCMAKE_SYSTEM_PROCESSOR=$ARCH \
+	      -DWITH_CUDA=OFF \
+	      -DBUILD_ANDROID_EXAMPLES=OFF \
+	      -DBUILD_DOCS=OFF \
+	      -DBUILD_PERF_TESTS=OFF \
+	      -DBUILD_TESTS=OFF \
+	      -DBUILD_ANDROID_PROJECTS=OFF \
+              -DBUILD_ANDROID_STL="libc++_shared"
```

Comment:
Try adding `android-libcxx` to the build requirements (and run requirements)

Reply:
ok

Reply:
i  fixed the error above creating a symbolic link to libstdc++.so `sudo ln -s /usr/lib/x86_64-linux-gnu/libstdc++.so.6 /usr/lib/x86_64-linux-gnu/libstdc++.so`

now we have other errors, but i go test add `android-libcxx ` like you sugest, to see if clang looks to c++ lib in local env

Reply:
ok, dont work, the only i find to bypass this is creating a symbolic link

Reply:
> Try adding `android-libcxx` to the build requirements (and run requirements)

ok, dont work, the only i find to bypass this is creating a symbolic link like:

> sudo ln -s /usr/lib/x86_64-linux-gnu/libstdc++.so.6 /usr/lib/x86_64-linux-gnu/libstdc++.so

Reply:
This will not work. You can't use an x86_64 lib on arm. It must link to libstdc++_shared.so that's added when android-libcxx is installed.

Reply:
@frmdstryr  how can i link for opencv look for c++ in `/android/$arch/lib/libc++_shared.so` inside env ?

Reply:
i think `-DBUILD_ANDROID_STL="libc++_shared" ` should work, but dont

Reply:
adding `    export LDSHARED="$CXX -shared"` no sucess to

Reply:
result always 

```
/home/rodgomesc/Android/Sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++       CMakeFiles/cmTC_cd820.dir/testCXXCompiler.cxx.o  -o cmTC_cd820 
    /usr/bin/ld: não foi possível localizar -lstdc++
    clang++: error: linker command failed with exit code 1 (use -v to see invocation)
    CMakeFiles/cmTC_cd820.dir/build.make:86: recipe for target 'cmTC_cd820' failed
    make[1]: *** [cmTC_cd820] Error 1
    make[1]: Leaving directory '/home/rodgomesc/miniconda3/conda-bld/android-opencv_1543074730513/work/build/CMakeFiles/CMakeTmp'
    Makefile:121: recipe for target 'cmTC_cd820/fast' failed
    make: *** [cmTC_cd820/fast] Error 2
    
```

Reply:
If you change the ARCHES back to (arm, arm64) does it get past this?

Reply:
I cloned and built and didn't get the same error, but it got stuck on another part after compiling about 30%.

Reply:
> I cloned and built and didn't get the same error, but it got stuck on another part after compiling about 30%.

is libjasper? i think i will need to create android-libjasper from  [https://github.com/mdadams/jasper/tree/master/src/libjasper](https://github.com/mdadams/jasper/tree/master/src/libjasper)

Reply:
> If you change the ARCHES back to (arm, arm64) does it get past this?

i will test it

Reply:
> I cloned and built and didn't get the same error, but it got stuck on another part after compiling about 30%.

I think I mentioned that you were talking about Ubuntu 16.04, some time ago, if you did not upgrade yet, and you managed to compile up to 30%, it was probably changes in the system version libraries, I'll wait for you to respond and if it's correct, I'll mount a virtual machine with ubuntu 16.04

Reply:
> If you change the ARCHES back to (arm, arm64) does it get past this?

is already seted only this archs
`export ARCHS=("arm arm64")`

it's normaly i see this path pointing to `x86_84`?
> /home/rodgomesc/Android/Sdk/ndk-bundle/toolchains/llvm/prebuilt/linux-x86_64/bin/clang++       > CMakeFiles/cmTC_cd820.dir/testCXXCompiler.cxx.o  -o cmTC_cd820 
>     /usr/bin/ld: não foi possível localizar -lstdc++ `